site/content/blog/2020-01-08-incorporating-golangci-lint.md

```diff
@@ -0,0 +1,140 @@
+WARN [runner] Can't run linter goanalysis_metalinter: structcheck: analysis skipped: errors in package: [/home/agniva/play/go/src/github.com/mattermost/mattermost-server/wsapi/api.go:7:2: could not import github.com/mattermost/mattermost-server/v5/app (/home/agniva/play/go/src/github.com/mattermost/mattermost-server/app/channel.go:1889:64: missing ',' before newline in argument list)] 
+```
+
+Compare that to the `go vet` output:
+
+```
+# github.com/mattermost/mattermost-server/v5/app
+app/channel.go:1889:64: syntax error: unexpected newline, expecting comma or )
+# github.com/mattermost/mattermost-server/v5/app
+vet: app/channel.go:1889:64: missing ',' before newline in argument list (and 10 more errors)
+```
+
+The warnings which should have been an error already had an [issue](https://github.com/golangci/golangci-lint/issues/866). We filed another [one](https://github.com/golangci/golangci-lint/issues/886) for the poor syntax errors.
+
+Lastly, we just wanted to touch upon something that might be surprising to new Gophers out there. Mattermost has a very thin enterprise layer which gets built by including the directory as a symlink from inside the open-source repo.
+
+While running golangci-lint with the enterprise layer included, we observed that it was failing to run the checks inside the symlink. This is due to the fact that the Go toolchain does not work very well with symlinks. As a simple fix, we added another [command](https://github.com/mattermost/mattermost-server/blob/f672eb729103ef0c8512a3facb48d44c386cd00a/Makefile#L163) to run that directory specifically.
+
+#### Future Work
+
+Now that we have a base in place, our plan is to add more linters. Here is a short wish-list for the near term, roughly in the order of priority:
+__errcheck__: This got missed out in our initial cut because the work involved was too big. It is one of the most common mistakes in Go code, and extremely important that these get caught.
```

Comment:
Could you separate each of these sections into paragraphs ? The markdown combines all of them into a single para, which becomes unreadable.

Reply:
Added bullet point for each of them